### PR TITLE
Handle late error when exception handler is active

### PR DESCRIPTION
**Pull request by mvorisek**
steps to reproduce:

1. output some unwatend output (`echo 'xx';`)
2. throw some exception

test for this should be added before merge

### CODE REVIEW

File: src/App.php [hunk 1]
```diff
@@ -1100,14 +1100,24 @@ protected function outputResponse(string $data, array $headers): void
 
         foreach (ob_get_status(true) as $status) {
             if ($status['buffer_used'] !== 0) {
-                throw new LateOutputError('Unexpected output detected');
+                $lateError = new LateOutputError('Unexpected output detected');
+                if ($this->catch_exceptions) {
+                    $this->caughtException($lateError);
```
Comment (mvorisek) [inline src/App.php:1105]: should terminate (and if not overrode it always will), if it does not, throw line below save us

**Comment by PhilippGrashoff** [comment]
Thumbs up for this one, very sensible.

**Review by abbadon1334 — APPROVED** [review]
Preapprove it, agree to add a test, we need it to check for PSR7

**Comment by mvorisek** [comment]
Merging, impossible to unit test as error/exception handler cannot be removed easily, as we do not know if the latest handler is the one set from the App - https://www.php.net/manual/en/function.restore-error-handler.php

Late errors can be tested with Behat only.

**Pull request merged**